source/sliding_puzzle.py

```diff
@@ -124,7 +139,88 @@ def _join(self, row: List[PuzzlePiece]) -> List[str]:
     def draw(self) -> str:
         """Draw the puzzle in its current state."""
         output = []
-        for row in self.pieces:
+        for row in self.rows:
             output.extend(self._join(row))
 
         return '\n'.join(output)
+
+    def shuffle(self) -> None:
+        """Shuffle the puzzle by randomizing the order of the pieces"""
+        for row in self.rows:
+            random.shuffle(row)
```

Comment:
You should add one final shuffle to shuffle all the rows

Reply:
ok, good idea